src/OrchardCore.Modules/OrchardCore.Users/Startup.cs

```diff
@@ -99,7 +99,7 @@ public override void ConfigureServices(IServiceCollection services)
             services.AddScoped<IDataMigration, Migrations>();
 
             services.AddScoped<IUserService, UserService>();
-            services.AddScoped<IUserClaimsPrincipalFactory<IUser>, DefaultUserClaimsPrincipalFactory>();
+            services.AddScoped<IUserClaimsPrincipalFactory<User>, DefaultUserClaimsPrincipalFactory>();
```

Comment:
Changing that indeed fixes https://github.com/OrchardCMS/OrchardCore/issues/2727, but it doesn't do what you think it does:

`DefaultUserClaimsPrincipalFactory` will be completely ignored since `UserManager<IUser>` (registered by `services.AddIdentity<IUser, IRole>()` will resolve `IUserClaimsPrincipalFactory<IUser>`, but not `IUserClaimsPrincipalFactory<User>`.

Concretely, this means `DefaultUserClaimsPrincipalFactory` is never used: if it was, I'm sure you'd get an exception saying it can't instantiate it due to `UserManager<User>` not being registered.